kernel/src/common/copy_slice.rs

```diff
@@ -0,0 +1,50 @@
+//! A wrapper around an immutable slice of `Copy` elements
+//! that provides access via a mutable slice
```

Comment:
As it's written, I'm skeptical this is safe. Here's one example:

```rust
#[derive(Copy,Clone)]
enum Foo {
  Num(usize),
  Ref(&'a usize)
}

let myslice: CopySlice<Foo> = ...;

let Ref(x) = myslice.as_slice()[0]; // `x` is a `&'static usize`

myslice.as_mut()[0] = Num(0xdeadbeef);

// now `x` is actually a reference to `0xdeadbeef`. oops...
```

However, I _think_ what you're trying to get at is a (more ergonomic) version of something like `&[Cell<T>]`, which can be shared (it's not a mutable reference) and each element can be modified since elements are `Cell`s, but the index operation gives you a `Cell<T>` rather a `T`, so you can never get multiple references to the same actual memory of an element:

```rust
myslice[0].get();
myslice[0].set(1234);
```

This _may_ be sufficient for what you care about in which case, despite the slightly less ergonomic type it might be better to just use `&[Cell<T>]`, or you may be able to write a similar wrapper type that implements the same semantics and hides some of the details behind the `Index` and `IndexMut` traits.